app/src/main/java/com/concordium/wallet/ui/cis2/TokenDetailsActivity.kt

```diff
@@ -131,15 +131,18 @@ class TokenDetailsActivity : BaseActivity() {
         }
     }
 
-    private fun setOwnership(tokenMetadata: TokenMetadata, token: Token) {
+    private fun setBalance(token: Token, tokenMetadata: TokenMetadata) {
         if (tokenMetadata.unique) {
+            binding.includeAbout.balanceHolder.visibility = View.GONE
             binding.includeAbout.ownershipHolder.visibility = View.VISIBLE
-            binding.includeAbout.ownership.text =
-                if (token.totalBalance != BigInteger.ZERO) {
-                    getString(R.string.cis_owned)
-                } else {
-                    getString(R.string.cis_not_owned)
-                }
+            binding.includeAbout.ownership.text = if (token.totalBalance != BigInteger.ZERO)
+                getString(R.string.cis_owned)
+            else
+                getString(R.string.cis_not_owned)
+        } else {
+            binding.includeAbout.ownershipHolder.visibility = View.GONE
+            binding.includeAbout.balanceHolder.visibility = View.VISIBLE
+            binding.includeAbout.balance.text = token.totalBalance.toString()
```

Comment:
We don't actually want to show the balance here, AFAIK, because it is displayed in the header of the view. :thinking: